static/js/hoc/withPostList_test.js

```diff
@@ -0,0 +1,218 @@
+
+        assert.equal(inner.find("InfiniteScroll").props().hasMore, !!afterUrl)
+      })
+    })
+
+    describe("loadMore", () => {
+      let pagination
+
+      beforeEach(() => {
+        pagination = makePagination()
+      })
+
+      it("doesn't do anything if canLoadMore is not set", async () => {
+        const { inner } = await render(
+          {},
+          {
+            canLoadMore: false,
+            pagination:  pagination
+          }
+        )
+        const props = inner.find("InfiniteScroll").props()
+        await props.loadMore()
+        sinon.assert.notCalled(loadPosts)
+      })
+      ;[true, false].forEach(hasParams => {
+        it(`${
+          hasParams ? "uses" : "doesn't use"
+        } query parameters from the URL for the pagination`, async () => {
+          const { inner } = await render(
+            {},
+            {
+              canLoadMore: true,
+              pagination:  pagination,
+              location:    {
+                search: hasParams ? "?sort=some_sort&after=different_after" : ""
+              }
+            }
+          )
+          const props = inner.find("InfiniteScroll").props()
+          await props.loadMore()
+          assert.equal(loadPosts.callCount, 1)
+          sinon.assert.calledWith(loadPosts, {
+            after: pagination.after,
+            count: pagination.after_count,
+            ...(hasParams ? { sort: "some_sort" } : {})
+          })
+        })
+      })
+    })
+
+    describe("PostList", () => {
+      it("has posts", async () => {
+        const { inner } = await render()
+        const props = inner.find("PostList").props()
+
+        assert.deepEqual(props.posts, postList)
+      })
+
+      it("toggles an upvote", async () => {
+        const post = makePost()
+        const innerStub = helper.sandbox.stub()
+        const stub = helper.sandbox.stub(apiActions, "toggleUpvote")
+        stub.get(() =>
+          R.curry(async (dispatch, post) => innerStub(dispatch, post))
+        )
+        const { inner } = await render()
+        const props = inner.find("PostList").props()
+        await props.toggleUpvote(post)
+
+        assert.equal(innerStub.callCount, 1)
+        assert.equal(innerStub.firstCall.args[1], post)
+      })
+      ;[
+        ["reportPost", "showReportPost"],
+        ["removePost", "showRemovePost"]
+      ].forEach(([prop, showProp]) => {
+        [true, false].forEach(showPropValue => {
+          it(`${
+            showProp ? "passes" : "doesn't pass"
+          } through ${prop} when ${String(showProp)}=${String(
+            showPropValue
+          )}`, async () => {
+            const { inner } = await render(
+              {},
+              {
+                [prop]:     prop,
+                [showProp]: showProp
+              }
+            )
+            assert.equal(
+              inner.find("PostList").props()[prop],
+              showProp ? prop : null
+            )
+          })
+        })
+      })
```

Comment:
This is kind of hard to read. I think as a general rule we should avoid nested loops for declaring test cases. I think it's better just to flatten it out. Something like this:

```javascript
    ;[
        ["reportPost", "showReportPost", true, true],
        ["reportPost", "showReportPost", false, false],
        ["removePost", "showRemovePost", true, true],
        ["removePost", "showRemovePost", false, false]
      ].forEach(([prop, showProp, showPropValue, shouldBePassed]) => {
        it(`${shouldIf(showProp)} pass down ${prop} prop when ${showProp}=${String(
          showPropValue
        )}`, async () => {
          const { inner } = await render(
            {},
            {
              [prop]:     prop,
              [showProp]: showPropValue
            }
          )
          assert.equal(
            notNil(inner.find("PostList").prop(prop)),
            shouldBePassed
          )
        })
      })
```